app/controllers/games_controller.rb

```diff
@@ -30,6 +30,13 @@ def update
     redirect_to root_path
   end
 
+  def forfeit
+    game = Game.find(params[:id])
+    return render_not_found(:unauthorized) if @game.white_player_id.present? && @game.black_player_id.present?
```

Comment:
Not understanding this line.

They won't be able to forfeit if the game has 2 players?

Also, `@game` here will be nil. Would need to be `game`.